crates/terminal/src/terminal.rs

```diff
@@ -1727,20 +1725,26 @@ impl Terminal {
         self.task.as_ref()
     }
 
-    pub fn wait_for_completed_task(&self, cx: &App) -> Task<()> {
-        if let Some(task) = self.task() {
-            if task.status == TaskStatus::Running {
-                let completion_receiver = task.completion_rx.clone();
-                return cx.spawn(|_| async move {
-                    let _ = completion_receiver.recv().await;
-                });
+    pub fn wait_for_completed_task(&self, cx: &App) -> Task<i32> {
+        let Some(task) = self.task() else {
+            return Task::ready(0);
+        };
+
+        match task.status {
+            TaskStatus::Completed { exit_code } => Task::ready(exit_code),
+            TaskStatus::Unknown => Task::ready(0),
+            TaskStatus::Running => {
+                let rx = task.completion_rx.clone();
+                cx.spawn(|_| async move { rx.recv().await.unwrap_or(-1) })
             }
         }
-        Task::ready(())
     }
 
     fn register_task_finished(&mut self, error_code: Option<i32>, cx: &mut Context<'_, Terminal>) {
```

Comment:
This is only called with `error_code` set to `None` when the terminal that the task is running in is dropped (or rather, alacritty emits an Exit event). Thus I don't think we should be assuming that the return value here is `0`. Ideally we'd propagate the lack of exit code upwards to the channel.